nautobot/ipam/models.py

```diff
@@ -596,6 +615,30 @@ def save(self, *args, **kwargs):
             parent = max(supernets, key=operator.attrgetter("prefix_length"))
             self.parent = parent
 
+        # Validate that creation of this prefix does not create an invalid parent/child relationship
+        if self.parent and self.parent.type != constants.PREFIX_ALLOWED_PARENT_TYPES[self.type]:
+            err_msg = f"{self.type.title()} prefixes cannot be created in {self.parent.type} prefixes"
```

Comment:
This error is a tad ambiguous. Perhaps something like?
```suggestion
            err_msg = f"{self.type.title()} prefixes cannot be children of {self.parent.type.title()} prefixes"
```